controllers/binding_controller_test.go

```diff
@@ -133,68 +227,81 @@ var _ = Describe("bindingController", func() {
-			BeforeEach(func() {
-				bindingName = "delete-binding-go-error"
-				fakeRabbitMQClient.DeleteBindingReturns(nil, errors.New("some error"))
-			})
-
-			It("raises an event to indicate a failure to delete", func() {
-				Expect(client.Delete(ctx, &binding)).To(Succeed())
-				Consistently(func() bool {
-					err := client.Get(ctx, types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace}, &topology.Binding{})
-					return apierrors.IsNotFound(err)
-				}, 5).Should(BeFalse())
-				Expect(observedEvents()).To(ContainElement("Warning FailedDelete failed to delete binding"))
-			})
+				return binding.Status.Conditions
+			}, 10*time.Second, 1*time.Second).Should(ContainElement(MatchFields(IgnoreExtras, Fields{
+				"Type":    Equal(topology.ConditionType("Ready")),
+				"Reason":  Equal("FailedCreateOrUpdate"),
+				"Status":  Equal(corev1.ConditionFalse),
+				"Message": Not(ContainSubstring("not allowed to reference")),
+			})))
 		})
+	})
 
-		When("the RabbitMQ Client successfully deletes a binding", func() {
-			BeforeEach(func() {
-				bindingName = "delete-binding-success"
-				fakeRabbitMQClient.DeleteBindingReturns(&http.Response{
-					Status:     "204 No Content",
-					StatusCode: http.StatusNoContent,
-				}, nil)
-			})
+	When("a binding references a cluster that allows all namespaces", func() {
+		JustBeforeEach(func() {
+			bindingName = "test-binding-allowed-when-allow-all"
+			binding = topology.Binding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      bindingName,
+					Namespace: "prohibited",
+				},
+				Spec: topology.BindingSpec{
+					RabbitmqClusterReference: topology.RabbitmqClusterReference{
+						Name:      "allow-all-rabbit",
+						Namespace: "default",
+					},
+				},
+			}
+		})
+		It("should be created", func() {
+			Expect(client.Create(ctx, &binding)).To(Succeed())
+			EventuallyWithOffset(1, func() []topology.Condition {
+				_ = client.Get(
+					ctx,
+					types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace},
+					&binding,
+				)
 
-			It("raises an event to indicate a successful deletion", func() {
-				Expect(client.Delete(ctx, &binding)).To(Succeed())
-				Eventually(func() bool {
-					err := client.Get(ctx, types.NamespacedName{Name: binding.Name, Namespace: binding.Namespace}, &topology.Binding{})
-					return apierrors.IsNotFound(err)
-				}, 5).Should(BeTrue())
-				Expect(observedEvents()).To(SatisfyAll(
-					Not(ContainElement("Warning FailedDelete failed to delete binding")),
-					ContainElement("Normal SuccessfulDelete successfully deleted binding"),
-				))
-			})
+				return binding.Status.Conditions
+			}, 10*time.Second, 1*time.Second).Should(ContainElement(MatchFields(IgnoreExtras, Fields{
+				"Type":    Equal(topology.ConditionType("Ready")),
+				"Reason":  Equal("FailedCreateOrUpdate"),
+				"Status":  Equal(corev1.ConditionFalse),
+				"Message": Not(ContainSubstring("not allowed to reference")),
+			})))
```

Comment:
When the binding references a cluster that allows all namespaces, it should create the binding, and eventually, the status condition `Ready` should be `True`, `Reason` should be `SuccessfulCreateOrUpdate`.

What's the reasoning behind having `Ready` condition to `False` if we expect the binding creation to succeed?